src/main/java/org/minions/devfund/sergio/Squarelotron.java

```diff
@@ -0,0 +1,212 @@
+        final int[][] auxArray = new int[size][size];
+        for (int i = 0; i < size; i++) {
+            for (int j = 0; j < size; j++) {
+                if (ringCheck(i, j, ring)) {
+                    auxArray[i][j] = squarelotronArray[size - 1 - i][j];
+                } else {
+                    auxArray[i][j] = squarelotronArray[i][j];
+                }
+            }
+        }
+        return auxArray;
+    }
+
+    /**
+     * Calls the method that swaps the square by main diagonal.
+     *
+     * @param ring - The ring to work with.
+     * @return The processed Square.
+     */
+    public Squarelotron mainDiagonalFlip(int ring) {
+        Squarelotron squarelotronResult = new Squarelotron(size);
+        squarelotronResult.squarelotron = mainDiagonalFlip(squarelotronResult.squarelotron, ring);
+        return squarelotronResult;
+    }
+
+    /**
+     * Makes the square swap by main diagonal.
+     *
+     * @param squarelotronArray - The given square.
+     * @param ring              - The ring to work whit.
+     * @return The Swapped square.
+     */
+    public int[][] mainDiagonalFlip(final int[][] squarelotronArray, int ring) {
+        final int[][] auxArray = new int[size][size];
+        for (int i = 0; i < size; i++) {
+            for (int j = 0; j < size; j++) {
+                if (ringCheck(i, j, ring)) {
+                    auxArray[i][j] = squarelotronArray[j][i];
+                } else {
+                    auxArray[i][j] = squarelotronArray[i][j];
+                }
+            }
+        }
+        return auxArray;
+    }
+
+    /**
+     * Makes the rotation to right according the number of turns.
+     *
+     * @param turns - Number of turns.
+     * @return The processed Array.
+     */
+    public int[][] rotateEast(int turns) {
+        while (turns > 0) {
+            for (int i = 1; i <= getNumberOfRings(); i++) {
+                this.squarelotron = upsideDownFlip(this.squarelotron, i);
+                this.squarelotron = mainDiagonalFlip(this.squarelotron, i);
+            }
+            turns--;
+        }
+        return Arrays.copyOf(this.squarelotron, size);
+    }
+
+    /**
+     * Makes the rotation to left according the number of turns.
+     *
+     * @param turns - Number of turns.
+     * @return - The processed Array.
+     */
+    public int[][] rotateWest(int turns) {
+        while (turns < 0) {
+            for (int i = 1; i <= getNumberOfRings(); i++) {
+                this.squarelotron = mainDiagonalFlip(this.squarelotron, i);
+                this.squarelotron = upsideDownFlip(this.squarelotron, i);
+            }
+            turns++;
+        }
+        return Arrays.copyOf(this.squarelotron, size);
+    }
+
+    /**
+     * Makes the rotation according the value of the numberOfTurns, left or right.
+     *
+     * @param numberOfTurns - The number of turns.
+     */
+    public void rotateRight(int numberOfTurns) {
+        if (numberOfTurns >= 0) {
```

Comment:
You can improve this.
Instead of rotating all the time 
you can calculate the actual turn
if the turn is 4 nothing should happen
if the turn is 5 only it should rotate 1 time.